pkg/controller/nodes/nodes_controller.go

```diff
@@ -87,17 +100,39 @@ func (r *ReconcileNodes) Start(stop <-chan struct{}) error {
 	}
 }
 
+func (r *ReconcileNodes) onUpdate(nodeName string) error {
+	logger := r.logger.WithValues("node", nodeName)
+	logger.Info("node update notification received")
+
+	c, err := r.getCache()
+	if err != nil {
+		return err
+	}
+
+	err = r.updateNode(c, nodeName)
+	if err != nil {
+		return err
+	}
+
+	err = r.updateCache(c)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
You can return the error directly.

```suggestion
	return r.updateCache(c)
```